src/rgw/rgw_sal_rados.cc

```diff
@@ -3143,80 +3199,92 @@ int RadosRole::read_name(const DoutPrefixProvider *dpp, optional_yield y)
     auto iter = bl.cbegin();
     decode(nameToId, iter);
   } catch (buffer::error& err) {
-    ldpp_dout(dpp, 0) << "ERROR: failed to decode role name from Role pool: " << name << dendl;
+    ldpp_dout(dpp, 0) << "ERROR: failed to decode role name from Role pool: " << info.name << dendl;
     return -EIO;
   }
-  id = nameToId.obj_id;
+  info.id = nameToId.obj_id;
   return 0;
 }
 
 int RadosRole::read_info(const DoutPrefixProvider *dpp, optional_yield y)
 {
   auto obj_ctx = store->svc()->sysobj->init_obj_ctx();
-  std::string oid = get_info_oid_prefix() + id;
+  std::string oid;
+
+  oid = info.id;
+  ldpp_dout(dpp, 0) << "INFO: oid in read_info is: " << oid << dendl;
```

Comment:
This should probably be removed (or at least raised > 0)

Reply:
done.